index.js

```diff
@@ -1,18 +1,74 @@
 const readline = require('readline');
 const rl = readline.createInterface(process.stdin, process.stdout);
-
+let minNum = 1
+let maxNum = 100
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     rl.question(questionText, resolve);
   });
 }
 
+// Original random function
+// function randomNum(min, max) {
+//   return Math.floor(Math.random() * (max - min + 1) + min)
+// }
+
+// binary search function
+function randomNum(min, max) {
+  return Math.floor((min + max) / 2)
+}
+
 start();
 
 async function start() {
   console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
   let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
+  //console.log('You entered: ' + secretNumber); (given with original code)
+  // Wait for user to think of a number and ask if random number is correct
+  let maxRange = await ask("But first, let's set your range! from 1 to ...? (Enter a number greater than 1)")
+  // making sure the user enters an actual number
+  if (isNaN(maxRange)) {
+    maxRange = await ask("That's not a valid response! Give me an actual number! (Enter a number greater than 1)")
+  }
```

Comment:
This is the right idea to ensure that people are giving you the proper inputs, but you'll want to put this inside of a loop rather than in an if statement. With it being in an if statement, the user can input the wrong answer twice and the second time it will accept it.